src/views/Restaurant.vue

```diff
@@ -2,49 +2,76 @@
   <div>
     <Header :id="restaurant.id" :title="restaurant.title" :slug="restaurant.slug" :logo="restaurant.logo" :banner="restaurant.banner"></Header>
     <HorizontalMenu title="Today's Special" :isLoading="isLoadingSpecials" :dishes="todaysSpecial"></HorizontalMenu>
-    <AccordionMenu title="Menu" :dishes="dishes" :courses="courses" :isLoading="isLoadingMenu"></AccordionMenu>
+    <div class="filtering">
+      <Filters
+        :dishes="dishes"
+        @filter-dish="filterDishes"
+        @clear-filter="clearFilter"
+      >
+      </Filters>
+      <SearchBar
+        :dishes="dishes"
+        @filter-dish="filterDishes"
+        @clear-filter="clearFilter"
+      >
+      </SearchBar>
+    </div>
+    <div class="filtered-items" ref="filteredItems" v-if="filteredDishes.length > 0">
+      <h2>Filtered Dishes</h2>
+      <Dish v-for="dish in filteredDishes" :key="dish.id"
+        :title="dish.title"
+        :image="dish.cover_image"
+        :description="dish.description"
+        :price="dish.price"
+        :ingredients="dish.ingredients"
+      ></Dish>
+    </div>
+    <div class="all-items" v-if="filteredDishes.length == 0">
+      <AccordionMenu :dishes="dishes" :courses="courses" :isLoading="isLoadingMenu" :activeByDefault="false"></AccordionMenu>
+    </div>
   </div>
 </template>
 <script>
 // Components
-import AccordionMenu from '@/components/AccordionMenu.vue'
-import Header from '@/components/Header.vue'
+import Dish from '@/components/Dish.vue'
+import AccordionMenu from '@/components/AccordionMenu.vue';
+import Header from '@/components/Header.vue';
 import HorizontalMenu from '../components/HorizontalMenu';
 
 // Additional
 import {GET_DISHES, GET_COURSES, GET_RESTAURANT, RESET_RESTAURANT, GET_TODAYS_SPECIAL} from '@/store/actions';
 import { mapState } from 'vuex';
+import SearchBar from '../components/SearchBar.vue';
+import Filters from '../components/Filters';
+
 
 export default {
   name: 'Restaurant',
-  components: { AccordionMenu, Header, HorizontalMenu },
+  components: { AccordionMenu, Header, HorizontalMenu, SearchBar, Filters, Dish },
   props: ["slug"],
   data () {
     return {
-      dishesSpecial: [],
-      categories: []
+      categories: [],
+      filteredDishes: []
     }
   },
   methods: {
-    async getSpecialDishes() {
-      try {
-        const data = await DishesService.getTodaysSpecial();
-        this.dishesSpecial = data;
-      } catch (error) {
-        console.log(error);
-      }  
-    },
     async getCategories() {
       try {
         const data = await CategoriesService.getAllCategories();
         this.categories = data;
```

Comment:
replace the two lines with `this.categories = await CategoriesService.getAllCategories();`

Reply:
Resolved